vernac/vernacentries.ml

```diff
@@ -44,6 +44,28 @@ let vernac_pperr_endline pp =
 
 (* Misc *)
 
+let there_are_pending_proofs ~pstate =
+  not Option.(is_empty pstate)
+
+let check_no_pending_proof ~pstate =
+  if there_are_pending_proofs ~pstate then
+    user_err Pp.(str "Command not supported (Open proofs remain)")
+
+let vernac_require_open_proof ~pstate f =
+  match pstate with
+  | Some pstate -> f ~pstate
+  | None -> user_err Pp.(str "Command not supported (No proof-editing in progress)")
+
+let get_current_or_global_context ~pstate =
+  match pstate with
+  | None -> let env = Global.env () in Evd.(from_env env, env)
+  | Some p -> Pfedit.get_current_context p
+
+let get_current_or_goal_context ~pstate glnum =
```

Comment:
Shouldn't this be `get_global_or_goal_context`?

Reply:
Indeed, fixed.